restutil/processor.go

```diff
@@ -0,0 +1,707 @@
+				return err
+			}
+			output.Deleted++
+			bar.Increment()
+		}
+		bar.FinishPrint("Done deletes")
+	}
+
+	return json.NewEncoder(os.Stdout).Encode(output)
+}
+
+func doCopy(sourceURL, destURL, id string) error {
+
+	su := sourceURL
+	if !strings.HasSuffix(su, "/") {
+		su = su + "/"
+	}
+
+	sreq, err := http.NewRequest("GET", fmt.Sprintf("%s%s", su, id), nil)
+	if err != nil {
+		return err
+	}
+	sreq.Header.Set("User-Agent", Useragent)
+	sresp, err := HttpClient.Do(sreq)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		io.Copy(ioutil.Discard, sresp.Body)
+		_ = sresp.Body.Close()
+	}()
+
+	if sresp.StatusCode != http.StatusOK {
+		return fmt.Errorf("error copying resource: %s", sresp.Status)
+	}
+
+	du := destURL
+	if !strings.HasSuffix(du, "/") {
+		du = du + "/"
+	}
+
+	dreq, err := http.NewRequest("PUT", fmt.Sprintf("%s%s", du, id), sresp.Body)
+	if err != nil {
+		return err
+	}
+	dreq.Header.Set("User-Agent", Useragent)
+	dreq.Header.Set("Content-type", "application/json")
+	dresp, err := HttpClient.Do(dreq)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		io.Copy(ioutil.Discard, dresp.Body)
+		_ = dresp.Body.Close()
+	}()
+	if dresp.StatusCode != http.StatusOK {
+		return fmt.Errorf("error copying resource: %s", dresp.Status)
+	}
+
+	return nil
+}
+
+func doDelete(destURL, id string) error {
+
+	du := destURL
+	if !strings.HasSuffix(du, "/") {
+		du = du + "/"
+	}
+
+	dreq, err := http.NewRequest("DELETE", fmt.Sprintf("%s%s", du, id), nil)
+	if err != nil {
+		return err
+	}
+	dresp, err := HttpClient.Do(dreq)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		io.Copy(ioutil.Discard, dresp.Body)
+		_ = dresp.Body.Close()
+	}()
+	if dresp.StatusCode != http.StatusOK {
+		return fmt.Errorf("error deleting resource: %s", dresp.Status)
+	}
+	return nil
+}
+
+func (rp *resourcePutter) putAllBinary(msgs <-chan *binaryMsg, failChan chan []byte, wg *sync.WaitGroup) {
+	for msg := range msgs {
+		putURL, err := generatePutURL(*msg.id, rp.baseURL)
+
+		if err != nil {
+			log.Errorf("generatePutURL Error=%v", err.Error())
+			if failChan != nil {
+				failChan <- []byte(*msg.id)
+			}
+			continue
+		}
+
+		var r io.Reader = *msg.body
+		err = rp.put(putURL.String(), r)
+
+		if err != nil {
+			log.Errorf("PUT putURL=%v, Error=%v", putURL, err.Error())
+			if failChan != nil {
+				failChan <- []byte(*msg.id)
+			}
+			(*msg.body).Close()
+			continue
+		}
+		(*msg.body).Close()
+	}
+	wg.Done()
+}
+
+func generatePutURL(id string, baseURL string) (putURL *url.URL, err error) {
+	if !strings.HasSuffix(baseURL, "/") {
+		baseURL = baseURL + "/"
+	}
+	putURL, err = url.Parse(baseURL)
+	if err != nil {
+		return
+	}
+	putURL, err = putURL.Parse(id)
+	return
+}
+
+func (rp *resourcePutter) putAll(resources <-chan resource, failChan chan []byte) error {
+	for r := range resources {
+		id := r[rp.idProperty]
+		idStr, ok := id.(string)
+		if !ok {
+			log.Info("unable to extract id property from resource, skipping")
```

Comment:
Yeah, I noticed that as well. I was going to fix it but wasn't sure if that was expected behaviour. If there was tests for it I could verify but project really had none.